app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AlbumsController < ApplicationController
+  before_action :get_album, only: [:show, :edit, :destroy, :upvote, :update]
+
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  def index
+    @albums = Album.order(ranked: :desc)
+  end
+
+  def show
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params)
+    if @album.save
```

Comment:
Because `Album.create` already calls `save` as part of doing its thing, we don't need to call it again here. To check whether or not `@album` as saved to the database we can use `if @album.persisted?` instead.